Automattic/wp-calypso
# Hosting Command Palette: Explore filter prop for improving search results

Related to https://github.com/Automattic/dotcom-forge/issues/4836 and https://github.com/Automattic/wp-calypso/pull/85228#issuecomment-1859302315

## Proposed Changes

* This PR uses the `filter` prop from the `cmd` to match whatever is entered in the search field to the label or `searchLabel` and display only these commands when a user is searching for commands.

<img width="862" alt="Screenshot 2023-12-18 at 3 02 19 PM" src="https://github.com/Automattic/wp-calypso/assets/25575134/9db062ed-6592-45ea-9dcf-61b2e18be8af">

## Testing Instructions

* Navigate to Calypso Live Link
* Press `cmd + k` on macOS or `ctrl + k` on Windows to trigger command palette 
* Type `reader` in the search field and observe that the command that comes up is `Open reader`
* Type `edge` in the search field and observe that the commands that are displayed  have `edge` in the command label or in their `searchLabel`

## Pre-merge Checklist
